library/std/src/sys/unix/fs.rs

```diff
@@ -1576,15 +1576,34 @@ mod remove_dir_impl {
 #[cfg(not(any(all(target_os = "macos", target_arch = "x86_64"), target_os = "redox")))]
 mod remove_dir_impl {
```

Comment:
I think it makes sense to move everything that uses `-at` syscalls to a separate module. `fs.rs` already is very bloated.
I called it `dir_fd` in #88731